src/lib.rs

```diff
@@ -49,6 +49,50 @@ pub enum Error {
     InvalidId,
 }
 
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            #[cfg(feature = "remote_endpoint")]
+            Error::ChannelSend => write!(
+                f,
+                "event_manager: failed to send message to remote endpoint."
```

Comment:
I think the convention is not to add a full stop at the end of the error message in the `Display` implementation. This also makes it easier to generate composite messages like you propose below.

Reply:
sure, will fix it.